File: app/src/main/java/com/apps/adrcotfas/goodtime/BL/CurrentSession.java
/*
 * Copyright 2016-2019 Adrian Cotfas
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is
 * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */

package com.apps.adrcotfas.goodtime.BL;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

/**
 * Stores the session remaining duration, type and timer state.
 * @see GoodtimeApplication
 */
public class CurrentSession {
    private final MutableLiveData<Long> mDuration = new MutableLiveData<>();
    private final MutableLiveData<TimerState> mTimerState = new MutableLiveData<>();
    private final MutableLiveData<SessionType> mSessionType = new MutableLiveData<>();
    private final MutableLiveData<String> mLabel = new MutableLiveData<>();

    public CurrentSession(long duration) {
        this.mDuration.setValue(duration);
        this.mTimerState.setValue(TimerState.INACTIVE);
        this.mSessionType.setValue(SessionType.WORK);
        this.mLabel.setValue(PreferenceHelper.getCurrentSessionLabel().label);
    }

    public LiveData<Long> getDuration() {
        return mDuration;
    }

    public LiveData<TimerState> getTimerState() {
        return mTimerState;
    }

    public LiveData<SessionType> getSessionType() {
        return mSessionType;
    }

    public LiveData<String> getLabel() {
        return mLabel;
    }

    public void setDuration(long newDuration) {
        mDuration.setValue(newDuration);
    }

    public void setTimerState(TimerState state) {
        mTimerState.setValue(state);
    }

    public void setSessionType(SessionType type) {
        mSessionType.setValue(type);
    }

    public void setLabel(String label) {
        mLabel.setValue(label);
    }
}

File: app/src/main/java/com/apps/adrcotfas/goodtime/bl/CurrentSession.kt
/*
 * Copyright 2016-2019 Adrian Cotfas
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is
 * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */
package com.apps.adrcotfas.goodtime.bl

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData

/**
 * Stores the session remaining duration, type and timer state.
 * @see GoodtimeApplication
 */
class CurrentSession(durationInput: Long, label: String) {

    private val _duration = MutableLiveData<Long>()
    private val _timerState = MutableLiveData<TimerState>()
    private val _sessionType = MutableLiveData<SessionType>()
    private val _label = MutableLiveData<String>()

    init {
        _duration.value = durationInput
        _timerState.value = TimerState.INACTIVE
        _sessionType.value = SessionType.WORK
        _label.value = label
    }

    val duration: LiveData<Long>
        get() = _duration
    val timerState: LiveData<TimerState>
        get() = _timerState
    val sessionType: LiveData<SessionType>
        get() = _sessionType
    val label: LiveData<String>
        get() = _label

    fun setDuration(newDuration: Long) {
        _duration.value = newDuration
    }

    fun setTimerState(state: TimerState) {
        _timerState.value = state
    }

    fun setSessionType(type: SessionType) {
        _sessionType.value = type
    }

    fun setLabel(label: String) {
        _label.value = label
    }
}